src/mergeCoverage.js

```diff
@@ -47,9 +50,15 @@ exports.mergeCoverageAndGenerateReports =
     const config = loadConfig();
     const mergedCoverage = mergeCoverage(config);
 
-    for (const [reportType, directoryFromConfig] of Object.entries(
-      reportsTypeMap,
-    )) {
+    // We always want to produce the json-summary report because it is required
+    // for the validation phase.
+    const reportsToProduce = [
+      JSON_SUMMARY_REPORT_KEY,
+      ...config.output.additionalReports,
+    ];
+
+    for (const reportType of reportsToProduce) {
+      const directoryFromConfig = reportsTypeMap[reportType];
```

Comment:
🤔 

I'm of two minds:

On the one hand, it would reduce the amount of code. `reportsToProduce` would go away. Additionally, users could add new reports simply by adding to the `additionalReports` config, assuming that the new reports need no additional configuration.

On the other hand, it's a bit non-obvious (to me, at least) to state that `path.dirname(config.input.coverageSummaryPath)` should be the default output directory for all reports. Why would an input parameter also be a default output parameter?

I'd be more comfortable making the default output directory `config.mergeCoveragePath`, but then we would still need `[JSON_SUMMARY_REPORT_KEY]` in our `reportsTypeMap` because it wouldn't be using the default — because it's a coincidence that `config.mergeCoveragePath` and `path.dirname(config.input.coverageSummaryPath)` are the same.

Was the intent of your comment to simplify the code, or to make it easier to add more reports without code changes (i.e. only config changes)? I'm not sure we can have both.

Reply:
I've tried a different approach that simplifies the code and eliminates the need for a default value: I have combined the `additionalReports` and `generatedHtmlPath` into a single config object. This way consumers can add any additional reports they like without a code change. NB that you cannot specify the `json-summary` report via this mechanism. Its value is still hard-coded to `path.dirname(config.input.coverageSummaryPath)` because it's required by the tool for validation.

Changes in 2c169f0003ab9e611e116ffe738345877aeed952